utils/network.ts

```diff
@@ -18,14 +19,32 @@ export async function getDefaultNetworkId(): Promise<NetworkId> {
 	}
 }
 
+export const getGweiGasPriceOrMaxFee = (gasPriceObj?: GasPrice | null) => {
+	if (!gasPriceObj) return wei(0);
+	const { maxFeePerGas, gasPrice } = gasPriceObj;
+	if (gasPrice) {
+		return wei(gasPrice, GWEI_DECIMALS);
+	}
+	return wei(maxFeePerGas || 0, GWEI_DECIMALS);
+};
+
 export const getTransactionPrice = (
-	gasPrice: Wei | null,
+	gasPrice: GasPrice | null,
 	gasLimit: GasLimitEstimate,
-	ethPrice: Wei | null
+	ethPrice: Wei | null,
+	optimismLayerOneFee: Wei | null
 ) => {
 	if (!gasPrice || !gasLimit || !ethPrice) return null;
```

Comment:
maybe we can return 0? Not sure

Reply:
hmm yea,  I wonder how it renders. As you can see I didn't touch that line.  Ideally it should show some indication that we're not ready to show the price yet.

Reply:
```
const transactionFee = getTransactionPrice(
  gasPrice,
  gasLimitEstimate,
  ethPriceRate,
  optimismLayerOneFee
);
...
transactionFee != null
  ? `(${formatCurrency(selectedPriceCurrency.name, transactionFee, {
	  sign: selectedPriceCurrency.sign,
	  })})`
  : ''
```
At the call site we are handling `null` by not showing anything, which I guess is slightly better than showing 0 incorrectly